### PR TITLE
Added Sequences for issue #115

### PR DESCRIPTION
**Pull request by rcraggs**
Working code and demo in sample

I added a test but I will need to look into how roboelectric can simulate taps in order to add more tests.

### CODE REVIEW

File: sample/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -106,4 +106,6 @@
     <string name="example_show_for_description">Displays prompt over the FAB button for 7 seconds</string>
 
     <string name="title_activity_list_example" translatable="false">List Example</string>
+    <string name="example_sequence">Show an example of a sequence of promps</string>
```
Comment (sjwall) [inline sample/src/main/res/values/strings.xml]: Missing a `t` from `prompts`

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetSequence.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+                            mOnCompleteListener.onSequenceComplete();
+                        }
+                    }
+            }
+        });
+
+        mPrompts.add(prompt);
+    }
+
+    public void show() {
```
Comment (sjwall) [inline library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetSequence.java]: Add JavaDoc

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetSequence.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    /**
+     * Set the listener to listen with the action to call when the sequence ends
+     * @param listener the listener with the action to execute
+     */
+    public void setSequenceCompleteListener(@Nullable SequenceCompleteListener listener){
+        mOnCompleteListener = listener;
+    }
+
+    public void addPrompt(MaterialTapTargetPrompt prompt){
```
Comment (sjwall) [inline library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetSequence.java]: Add JavaDoc

File: library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetSequence.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+package uk.co.samuelwall.materialtaptargetprompt;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MaterialTapTargetSequence {
```
Comment (sjwall) [inline library/src/main/java/uk/co/samuelwall/materialtaptargetprompt/MaterialTapTargetSequence.java]: Add JavaDoc

**Comment by rcraggs** [comment]
thanks @sjwall I think, given that the `setExtraPromptStateChangeListener` on a PromptOption is only for sequences and shouldn't be used in general - then it would be better to name it accordingly. e.g. 
`setSequenceListener`

**Comment by sjwall** [comment]
Thank you for your work

**Pull request merged**